azurelinuxagent/ga/exthandlers.py

```diff
@@ -232,6 +232,16 @@ def get_exthandlers_handler(protocol):
     return ExtHandlersHandler(protocol)
 
 
+def list_agent_lib_directory(skip_agent_package=True):
+    for name in os.listdir(conf.get_lib_dir()):
+        path = os.path.join(conf.get_lib_dir(), name)
```

Comment:
call conf.get_lib_dir() once and reuse ?

Reply:
This is actually an atomic operation but good catch, this makes more sense, I'll implement this!